eq/server/config/resources.cpp

```diff
@@ -315,19 +315,23 @@ bool Resources::discover( ServerPtr server, Config* config,
         std::stringstream name;
         if( info.device == LB_UNDEFINED_UINT32 &&
             // VirtualGL display redirects to local GPU (see continue above)
-            !(info.flags & hwsd::GPUInfo::FLAG_VIRTUALGL) )
+            !(info.flags & hwsd::GPUInfo::FLAG_VIRTUALGL ))
         {
             name << "display";
         }
         else
         {
             name << "GPU" << ++gpuCounter;
-            if( info.flags & hwsd::GPUInfo::FLAG_VIRTUALGL &&
+            if( // When running under VirtualGL, GPUs that are not VNC virtual
+                // devices mustn't be interposed.
+                ( info.flags & ( hwsd::GPUInfo::FLAG_VIRTUALGL |
+                                 hwsd::GPUInfo::FLAG_VNC )) ==
+                    hwsd::GPUInfo::FLAG_VIRTUALGL &&
```

Comment:
I agree on that last comment. 

The crash happens only when running with vglrun. As a matter of fact, Equalizer does filter non GLX capable displays, because hwsd  discovery doesn't report them (gpu/glx/module.cpp:92) (unless you use vglrun as I explain below), so if you run eqPly on the VNC server without VirtualGL it will use the devices from :0 and ignore the VNC display (let's say :1), and you can't see anything.
Now, when vglrun is used, VirtualGL will report :1 as having the GLX extension (this is "works as design". However, if the display is added to exclude list, it will happen that in fact, it doesn't have it, making Equalizer crash later on. Notice that VirtualGL it no lying, because the extension is queried before the display is added to the exclude list.

So my answer is that it can be made clearer but not simpler. I was thinking of moving the condition to a separate function, but I'm not fully sure about the logic behind the comparison of info.device  against LB_UNDEFINED_UINT32 (which works for the virtual server created by ssh -X or vglconnect in a way that I don't understand because hwsd reports the :10.0 or whatever it is), for that reason I preferred to wait for your comments.